src/libraries/System.Collections.Immutable/src/System/Collections/ThrowHelper.cs

```diff
@@ -26,8 +26,8 @@ public static void ThrowIfNull(object arg, [CallerArgumentExpression(nameof(arg)
             throw new ArgumentNullException(paramName);
 
         [DoesNotReturn]
-        public static void ThrowKeyNotFoundException() =>
-            throw new KeyNotFoundException();
+        public static void ThrowKeyNotFoundException<TKey>(TKey key) =>
+            throw new KeyNotFoundException(SR.Format(SR.Arg_KeyNotFoundWithKey, key));
```

Comment:
This will not be a useful message for some types of key e.g. object - is that ok?

Reply:
It's what Dictionary does